govcd/access_control.go

```diff
@@ -361,6 +361,10 @@ func (adminCatalog *AdminCatalog) getAccessControlHeader(useTenantContext bool)
 
 // GetControlAccess read and returns the control access parameters from a VDC
 func (vdc *Vdc) GetControlAccess(useTenantContext bool) (*types.ControlAccessParams, error) {
+	if vdc.client == nil || vdc.Vdc == nil || vdc.Vdc.Name == "" {
+		return nil, fmt.Errorf("vdc struct must be initialized before using this method")
```

Comment:
let's be explicit what is missing in struct

Reply:
Changed.
I included a new function called `checkSanityVdcControlAccess` that is called at the top of `Vdc.SetControlAccess` and `Vdc.GetControlAccess` and sends accurate error messages now.

Reply:
great.